nsq/reader.py

```diff
@@ -665,10 +646,21 @@ def _redistribute_rdy_state(self):
                     logger.info('[%s:%s] idle connection, giving up RDY count', conn.id, self.name)
                     self._send_rdy(conn, 0)
 
+            conns = self.conns.values()
+
+            in_flight = [c for c in conns if c.in_flight]
```

Comment:
This might be further complicated because the `max_in_flight` variable below is a terrible name. Perhaps a better name would be `available_rdy`?